frappe/public/js/frappe/form/multi_select_dialog.js

```diff
@@ -1,110 +1,76 @@
-		/* Options: doctype, target, setters, get_query, action */
+	init: function (opts) {
+		/* Options: doctype, target, setters, get_query, action, add_filters_group, data_fields, primary_action_label */
 		$.extend(this, opts);
-
 		var me = this;
-		if(this.doctype!="[Select]") {
-			frappe.model.with_doctype(this.doctype, function(r) {
+		if (this.doctype != "[Select]") {
+			frappe.model.with_doctype(this.doctype, function () {
 				me.make();
 			});
 		} else {
 			this.make();
 		}
 	},
-	make: function() {
+	make: function () {
 		let me = this;
 
 		this.page_length = 20;
 		this.start = 0;
 
-		let fields = [
-			{
-				fieldtype: "Data",
-				label: __("Search Term"),
-				fieldname: "search_term"
-			},
-			{
-				fieldtype: "Column Break"
-			}
-		];
-		let count = 0;
-		if(!this.date_field) {
-			this.date_field = "transaction_date";
-		}
+		let fields = [];
 
-		// setters can be defined as a dict or a list of fields
-		// setters define the additional filters that get applied
-		// for selection
+		let header_fields = this.make_dialog_header();
 
-		// CASE 1: DocType name and fieldname is the same, example "customer" and "customer"
-		// setters define the filters applied in the modal
-		// if the fieldnames and doctypes are consistently named,
-		// pass a dict with the setter key and value, for example
-		// {customer: [customer_name]}
-
-		// CASE 2: if the fieldname of the target is different,
-		// then pass a list of fields with appropriate fieldname
+		fields = fields.concat(header_fields);
 
-		if($.isArray(this.setters)) {
-			for (let df of this.setters) {
-				fields.push(df, {fieldtype: "Column Break"});
-			}
-		} else {
-			Object.keys(this.setters).forEach(function(setter) {
-				fields.push({
-					fieldtype: me.target.fields_dict[setter].df.fieldtype,
-					label: me.target.fields_dict[setter].df.label,
-					fieldname: setter,
-					options: me.target.fields_dict[setter].df.options,
-					default: me.setters[setter]
-				});
-				if (count++ < Object.keys(me.setters).length) {
-					fields.push({fieldtype: "Column Break"});
-				}
-			});
+		if (this.add_filters_group) {
+			fields = fields.concat([
+				{
+					fieldtype: 'HTML',
+					fieldname: 'filter_area',
+				},
+			]);
 		}
 
 		fields = fields.concat([
-			{
-				"fieldname":"date_range",
-				"label": __("Date Range"),
-				"fieldtype": "DateRange",
-			},
-			{ fieldtype: "Section Break" },
 			{ fieldtype: "HTML", fieldname: "results_area" },
-			{ fieldtype: "Button", fieldname: "more_btn", label: __("More"),
-				click: function(){
+			{
+				fieldtype: "Button", fieldname: "more_btn", label: __("More"),
+				click: function () {
 					me.start += 20;
 					frappe.flags.auto_scroll = true;
 					me.get_results();
 				}
 			}
 		]);
 
+		if (this.data_fields) {
+			fields = fields.concat([{ fieldtype: "Section Break" }]);
+			fields = fields.concat(this.data_fields);
+
+		}
+
 		let doctype_plural = !this.doctype.endsWith('y') ? this.doctype + 's'
```

Comment:
Make a separate utility in `utils.js` and use it here

Reply:
done